src/common/AutoComplete.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useRef, useState } from "react";
+import "./AutoComplete.scss";
+import { GoalItem } from "@src/models/GoalItem";
+
+interface AutocompleteComponentProps {
+  data: GoalItem[];
+  onSuggestionClick: (suggestion: GoalItem) => void;
+  onInputChange: (value: string) => void;
+  inputvalue: string;
+  placeholder: string;
+}
+
+const AutocompleteComponent: React.FC<AutocompleteComponentProps> = ({
+  data,
+  onSuggestionClick,
+  onInputChange,
+  inputvalue,
+  placeholder,
+}) => {
+  const [autocompleteValue, setAutocompleteValue] = useState("");
```

Comment:
remove this state.

Reply:
@Tushar-4781 why remove the state. it is responsible for suggestion value.